apps/admin-server/src/components/document-uploader.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useEffect } from 'react';
+import { FieldValues, Path, UseFormReturn, useForm } from 'react-hook-form';
+import {
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from './ui/form';
+import { Input } from './ui/input';
+
+export const DocumentUploader: React.FC<{
+  form: UseFormReturn<any>;
+  fieldName: Path<FieldValues>;
+  onDocumentUploaded?: (documentObject: {url: string} ) => void;
+  documentLabel?: string;
+  allowedTypes?: string[];
+}> = ({ form, fieldName, onDocumentUploaded, allowedTypes, documentLabel = 'Document' }) => {
+  const [document, setDocument] = React.useState<{url: string, name: string}>();
+  const [documentUrl, setDocumentUrl] = React.useState<string>('');
+
+  function prepareDocument(document: any) {
+    const formData = new FormData();
+    formData.append('document', document);
+    formData.append('documentname', 'testName');
+    formData.append('description', 'testDescription');
```

Comment:
Waar is dit voor?

Reply:
Dit was ingesteld voor de afbeelding uploader, dus ik kreeg het idee dat het wel nodig was. Anders kunnen we het voor nu mergen en dat we het op een lijst zetten om later uit te zoeken?